Add HLSStream tests

dthaler opened the pull request:
Add tests for HLSStream.py to github CI.

The tests pass with PR #29 and #28 merged (verified in my fork in [this run](https://github.com/dthaler/orca-hls-utils/actions/runs/19002165638)).  They fail with them not merged (verified in my fork in [this run](https://github.com/dthaler/orca-hls-utils/actions/runs/19002220396)).

Addresses part of issue #18

valentina-s commented on .github/workflows/test_hlsstream.yml:
```diff
@@ -0,0 +1,46 @@
+
+on:
+  push:
+    branches: [ master, main ]
```
We have only master right now. Most probably we should rename it to main.

valentina-s commented on .github/workflows/test_hlsstream.yml:
```diff
@@ -0,0 +1,46 @@
+name: Test HLSStream
```
Rename as `test.yml` since eventually will be one worflow for all functionality

valentina-s commented on tests/test_hlsstream.py:
```diff
@@ -0,0 +1,459 @@
+    print("=" * 60)
+
+    # Default stream base for tests
+    default_stream_base = (
```
these can be defined as fixtures so that all tests use them while running pytest.

dthaler commented:
> I also wonder whether using `utcnow` will make the tests unpredictable in the future? I guess in order to test the streaming behavior one can do it with a fixed date and a mock stream example.

Until there is a mock stream, utcnow is essential to be able to test the case when one has to wait before more files will appear.

dthaler commented:
@valentina-s Thanks for the feedback, I haven't used python much before, just other languages.  I believe I've addressed your feedback on this PR now.  https://github.com/dthaler/orca-hls-utils/pull/6/checks shows the pytest tests passing in my fork.

dthaler commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,47 @@
+name: Test HLSStream
```
```suggestion
name: Test
```
